census/README.md

```diff
@@ -1,313 +1,322 @@
-   * Activate env `source activate single-tf`
+We host the Census files required for you to run this sample. Files are hosted in a Public Bucket.
+
+If you want to use local files directly, you can use the following commands:
 
+TensorFlow - not the Cloud ML Engine - handles reading from GCS, you can run all commands below using these environment variables. However, if your network is slow or unreliable, you may want to download the files for local training.
+
+```
+CENSUS_DATA=census_data
+mkdir $CENSUS_DATA
+gsutil cp gs://cloud-samples-data/ml-engine/census/data/adult.data.csv $CENSUS_DATA
+gsutil cp gs://cloud-samples-data/ml-engine/census/data/adult.test.csv $CENSUS_DATA
+```
 
-## Install dependencies
+* **Upload the data to a Google Cloud Storage bucket**
 
- * Install [gcloud](https://cloud.google.com/sdk/gcloud/)
- * Install the python dependencies. `pip install --upgrade -r requirements.txt`
+Cloud ML Engine works by using resources available in the cloud, so the training
+data needs to be placed in such a resource. For this example, we'll use [Google
+Cloud Storage], but it's possible to use other resources like [BigQuery]. Make a
+bucket (names must be globally unique) and place the data in there:
 
-# Single Node Training
-Single node training runs TensorFlow code on a single instance. You can run the exact
-same code locally and on Cloud ML Engine.
+```shell
+gsutil mb gs://your-bucket-name
+gsutil cp -r data/adult.data.csv gs://your-bucket-name/adult.data.csv
+gsutil cp -r data/adult.test.csv gs://your-bucket-name/adult.test.csv
+```
 
-## How to run the code
-You can run the code either as a stand-alone python program or using `gcloud`.
-See options below:
+<h1>Training</h1>
 
-### Using local python
-Run the code on your local machine:
+* **GCloud configuration:**
 
 ```
 export TRAIN_STEPS=1000
 DATE=`date '+%Y%m%d_%H%M%S'`
-export OUTPUT_DIR=census_$DATE
-rm -rf $OUTPUT_DIR
+export JOB_DIR=census_$DATE
+rm -rf $JOB_DIR
+export TRAIN_FILE=census_data/adult.data.csv
+export EVAL_FILE=census_data/adult.test.csv
 ```
 
-### Model location reuse
-Its worth calling out that unless you want to reuse the old model output dir,
-model location should be a new location so that old model doesn't conflict with new
-one. 
+* **Test locally:**
 
 ```
 python -m trainer.task --train-files $TRAIN_FILE \
-                       --eval-files $EVAL_FILE \
-                       --job-dir $OUTPUT_DIR \
-                       --train-steps $TRAIN_STEPS \
-                       --eval-steps 100
+    --eval-files $EVAL_FILE \
+    --job-dir $JOB_DIR \
+    --train-steps $TRAIN_STEPS \
+    --eval-steps 100
 ```
 
-### Using gcloud local
-Run the code on your local machine using `gcloud`. This allows you to mock
-running it on the cloud:
+* **Google Cloud ML Engine**
+
+* **GCloud configuration:**
 
 ```
-export TRAIN_STEPS=1000
 DATE=`date '+%Y%m%d_%H%M%S'`
-export OUTPUT_DIR=census_$DATE
-rm -rf $OUTPUT_DIR
+export JOB_DIR=census_$DATE
```

Comment:
here it is `JOB_DIR` but it becomes `GCS_JOB_DIR` below.

Reply:
I liked that he switched between JOB_DIR for testing locally and then GCS_JOB_DIR when running on the cloud to help distinguish the two, otherwise they have to overwrite JOB_DIR and people might think they did that already and skip that step. But, I could also see the point for consistency to match the parameters.

Reply:
While I was going through the samples, (copy/paste) starting from running locally to running in Cloud ML, I missed updating this flag couple of times,  Now I'm directly forced to update it and implying where this data belongs.